cli/azd/pkg/config/manager_test.go

```diff
@@ -39,3 +42,40 @@ func Test_SaveAndLoadEmptyConfig(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, existingConfig)
 }
+
+func Test_DirectoryPermissions(t *testing.T) {
+	if runtime.GOOS != "linux" && runtime.GOOS != "darwin" {
+		// Run this test only on Linux/macOS
+		return
+	}
+
+	getPermissions := func(t *testing.T, path string) fs.FileMode {
+		info, err := os.Stat(path)
+		require.NoError(t, err)
+
+		return info.Mode().Perm()
+	}
+
+	testDir, err := os.MkdirTemp(os.TempDir(), "azd_config_dir_test*")
```

Comment:
It's very common to create temp directories in tests, that `t *testing.T` offers: `testDir := t.TempDir()` as a convenience function